src/api/spec/controllers/concerns/triggerable_spec.rb

```diff
@@ -0,0 +1,173 @@
+require 'rails_helper'
+
+RSpec.describe Triggerable do
+  let(:fakes_controller) do
+    Class.new(ApplicationController) do
+      include Triggerable
+    end
+  end
+
+  let(:object) { fakes_controller.new }
+
+  let(:user) { create(:confirmed_user, login: 'foo') }
+  let(:project) { create(:project, name: 'project', maintainer: user) }
+  let(:package) { create(:package, name: 'package_trigger', project: project) }
+  let(:repository) { create(:repository, name: 'package_test_repository', architectures: ['x86_64'], project: project) }
+
+  before do
+    object.instance_variable_set(:@token, token)
+  end
+
+  def stub_params(project_name:, package_name:)
+    stubbed_params = ActionController::Parameters.new(project: project_name, package: package_name)
+    allow(object).to receive(:params).and_return(stubbed_params)
+  end
+
+  describe '#set_project' do
+    let(:token) { Token::Rebuild.create(user: user) }
+
+    before do
+      allow(Project).to receive(:get_by_name).and_return('some:remote:project')
+    end
+
+    it 'raises a not found for a remote project' do
+      stub_params(project_name: 'some:remote:project', package_name: package.name)
+      expect { object.set_project }.to raise_error(Project::Errors::UnknownObjectError)
+    end
+  end
+
+  describe '#set_package' do
+    let(:token) { Token::Service.create(user: user) }
+
+    it 'raises when package does not exist' do
+      stub_params(project_name: project.name, package_name: 'does-not-exist')
+      object.set_project
+      expect { object.set_package }.to raise_error(Package::Errors::UnknownObjectError)
+    end
+
+    context 'project with project-link and token that follows project-links' do
+      let(:token) { Token::Rebuild.create(user: user) }
+      let(:project_with_a_link) { create(:project, name: 'project_with_a_link', maintainer: user, link_to: project) }
+
+      it 'raises when package does not exist in link' do
+        stub_params(project_name: project_with_a_link.name, package_name: 'does-not-exist')
+        object.set_project
+
+        expect { object.set_package }.to raise_error(Package::Errors::UnknownObjectError)
+      end
+
+      it 'assigns linked package' do
+        stub_params(project_name: project_with_a_link.name, package_name: package.name)
+        object.set_project
```

Comment:
unless you are testing he methods `set_project`, `set_package` and `set_object_to_authorize`, you could as well stub them with `allow` and `receive`